### PR TITLE
Do refactor for docker-machine jobs

### PR DESCRIPTION
**Pull request by kiwik**
1. Add role "config-docker-machine" to compiled and install
   docker-machine from source code.
2. Change job logic in order to putting related code together
3. Fix cp to same path issue, because we have used upstream project,
   don't need to cp forked project into official path

Related-Bug: theopenlab/openlab#133

### CODE REVIEW

File: roles/config-docker-machine/tasks/main.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      name: config-golang
+  - name: install docker-machine from source
+    shell: |
+      set -ex
+
+      # Install bats
+      git clone https://github.com/sstephenson/bats.git '{{ ansible_user_dir }}/bats'
+      pushd '{{ ansible_user_dir }}/bats'
+      ./install.sh /usr/local
+      popd
```
Comment (liu-sheng) [inline roles/config-docker-machine/tasks/main.yaml]: bats is not related with docker-machine installation, it is only for running tests.

File: roles/config-docker-machine/tasks/main.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+---
+- name: Compiled and install docker-machine from source
+  block:
```
Comment (liu-sheng) [inline roles/config-docker-machine/tasks/main.yaml]: I don't think we need `block` here

File: roles/config-docker-machine/tasks/main.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      # Install bats
+      git clone https://github.com/sstephenson/bats.git '{{ ansible_user_dir }}/bats'
+      pushd '{{ ansible_user_dir }}/bats'
+      ./install.sh /usr/local
+      popd
+
+      # Install docker
+      apt-get update
+      apt-get install apt-transport-https ca-certificates
+      wget -qO- https://get.docker.com/ | sh
```
Comment (liu-sheng) [inline roles/config-docker-machine/tasks/main.yaml]: docker installation also looks a bit unrelated with docker-machine installation,.. but OK

File: roles/config-docker-machine/tasks/main.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+      # NOTE: backport fix of PR 4543 and PR 4545
+      git config --global user.email 'zuul@openlab.com'
+      git config --global user.name 'openlab'
+      git cherry-pick 9ec6729
+      git cherry-pick 7909ee3
```
Comment (liu-sheng) [inline roles/config-docker-machine/tasks/main.yaml]: maybe we can switch to use the latest release v0.16.0 and remove this workaround

Comment (kiwik) [reply roles/config-docker-machine/tasks/main.yaml]: Yes, I consider that too, will do it in separated patch.

File: roles/config-docker-machine/tasks/main.yaml (hunk 2 above)
Comment (kiwik) [reply roles/config-docker-machine/tasks/main.yaml]: OK

File: roles/config-docker-machine/tasks/main.yaml (hunk 3 above)
Comment (kiwik) [reply roles/config-docker-machine/tasks/main.yaml]: Putting docker and docker-machine installation together make sense, docker-machine tests depends on it.

File: roles/config-docker-machine/tasks/main.yaml (hunk 1 above)
Comment (kiwik) [reply roles/config-docker-machine/tasks/main.yaml]: OK, will move into job logic
